### DISCUSSION TITLE
Can I somehow add custom field to the Patient resource and parse it?

### DISCUSSION THREAD

**Discussion by dronSamogon** [FirelyTeam/firely-net-sdk#1835] [Q&A] [answered]
Hello everyone!

I want to add property, lets say; "customId", into Patient resource json.
Something like this:
`{
   "resourceType": "Patient",
  "id": "example",
  "customId": 123123,
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">\n\t\t\t<table>\n\t\t\t\t<tbody>\n\t\t\t\t\t<tr>\n\t\t\t\t\t\t<td>Name</td>\n\t\t\t\t\t\t<td>Peter James \n              <b>Chalmers</b> (&quot;Jim&quot;)\n            </td>\n\t\t\t\t\t</tr>\n\t\t\t\t\t<tr>\n\t\t\t\t\t\t<td>Address</td>\n\t\t\t\t\t\t<td>534 Erewhon, Pleasantville, Vic, 3999</td>\n\t\t\t\t\t</tr>\n\t\t\t\t\t<tr>\n\t\t\t\t\t\t<td>Contacts</td>\n\t\t\t\t\t\t<td>Home: unknown. Work: (03) 5555 6473</td>\n\t\t\t\t\t</tr>\n\t\t\t\t\t<tr>\n\t\t\t\t\t\t<td>Id</td>\n\t\t\t\t\t\t<td>MRN: 12345 (Acme Healthcare)</td>\n\t\t\t\t\t</tr>\n\t\t\t\t</tbody>\n\t\t\t</table>\n\t\t</div>"
  },
  "identifier" ...`

And I want to use Hl7.Fhir.Serialization.FhirJsonParser() to parse it. How can I make it?

**Comment by marcovisserFurore** [comment, marked as answer]
Hi @dronSamogon,

Adding extra properties to the FHIR json is not allowed: the FHIR standard is built for interoperability and changing the json schema will disturb this interoperability. You can extend a FHIR resource by profiling it: https://www.hl7.org/fhir/profiling.html.

But in your case you can also use the Patient property Identifier. This is a list of identifiers. In your case it looks like this:

```json

{ 
   "resourceType": "Patient", 
   "id": "example", 
   "identifier": [
      {
        "use": "usual",
        "system": "http://your.system/customid",
        "value": "123123"
      }
    ] 
}
```

**Reply by dronSamogon** [reply]
Thank you, Marco! I got that.
